docs/articles/semantic_search_news.md

```diff
@@ -0,0 +1,376 @@
+
+First, we'll define a schema for our news articles.
+
+```python
+# set up schema to accommodate our inputs
+class NewsSchema(Schema):
+    description: String
+    headline: String
+    release_timestamp: Timestamp
+    id: IdField
+```
+
+```python
+news = NewsSchema()
+```
+
+Next, to embed the characteristics of our text, we use a sentence-transformers model to create a `description_space` for news article descriptions and a `headline_space` for our headlines, and, finally, we encode each article's release date using a `recency_space`.
+
+```python
+# textual characteristics are embedded using a sentence-transformers model
+description_space = TextSimilaritySpace(
+    text=news.description, model="sentence-transformers/all-mpnet-base-v2"
+)
+headline_space = TextSimilaritySpace(
+    text=news.headline, model="sentence-transformers/all-mpnet-base-v2"
+)
+# release date is encoded using our recency embedding algorithm
+recency_space = RecencySpace(
+    timestamp=news.release_timestamp,
+    period_time_list=[
+        PeriodTime(timedelta(days=4 * YEAR_IN_DAYS), weight=1),
+        PeriodTime(timedelta(days=11 * YEAR_IN_DAYS), weight=2),
+    ],
+    negative_filter=0.0,
+)
+```
+
+To query our data, we'll need to create an index of our spaces...
+
+```python
+news_index = Index(spaces=[description_space, headline_space, recency_space])
+```
+
+...and set up a **simple query** and a **news query**.
+
+**Simple query** lets us use a search term to retrieve from both the headline and the description. Simple query also gives us the option to weight certain inputs' importance.
+
+```python
+simple_query = (
+    Query(
+        news_index,
+        weights={
+            description_space: Param("description_weight"),
+            headline_space: Param("headline_weight"),
+            recency_space: Param("recency_weight"),
+        },
+    )
+    .find(news)
+    .similar(description_space.text, Param("query_text"))
+    .similar(headline_space.text, Param("query_text"))
+    .limit(Param("limit"))
+)
+```
+
+**News query** will search our database using the vector for a specific news article. News query, like simple query, can be weighted.
+
+```python
+news_query = (
+    Query(
+        news_index,
+        weights={
+            description_space: Param("description_weight"),
+            headline_space: Param("headline_weight"),
+            recency_space: Param("recency_weight"),
+        },
+    )
+    .find(news)
+    .with_vector(news, Param("news_id"))
+    .limit(Param("limit"))
+)
+```
+
+Next we parse our dataframe,...
+
+```python
+dataframe_parser = DataFrameParser(
+    schema=news,
+    mapping={news.release_timestamp: "date", news.description: "short_description"},
+)
+```
+
+...create an InMemorySource object to hold the user data in memory, and set up our executor (with our article dataset and index) so that it takes account of context data. The executor creates vectors based on the index's grouping of Spaces.
```

Comment:
actually Source doesn't really hold the data, the concept doesn't really make sense in this interactive notebook environment, it is here for compatibility with other modes. Let's say it accepts the data and it is stored in an InMemoryVectorDatabase (which is implicit in the code).